### PR TITLE
chore: Add debugger CE changes

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

- Support Debugger logs for ModuleInputsEntity
- Split the EntityLink component to add ModuleInputsEntityLink
- Add a temporary icon for ModuleInputEntity in the debugger until designs are ready.  (EE PR change )

CE changes related to https://github.com/appsmithorg/appsmith-ee/pull/2836

#### PR fixes following issue(s)
Fixes #

#### Type of change

- Chore

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/components/editorComponents/Debugger/entityTypeLinkMap.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
+import { PluginType } from "entities/Action";
```
Comment (ApekshaBhosale) [inline app/client/src/ce/components/editorComponents/Debugger/entityTypeLinkMap.tsx:1]: is this required to move entire component to ce? what kind oc hangs are you planning to do for ee? @Rishabh-Rathod

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/ce/components/editorComponents/Debugger/entityTypeLinkMap.tsx (hunk 1 above)
Comment (Rishabh-Rathod) [reply app/client/src/ce/components/editorComponents/Debugger/entityTypeLinkMap.tsx:1]: @ApekshaBhosale No it is not required. I will split this and only move actually required code to the CE folder. Thanks for the heads-up.

File: app/client/src/entities/DataTree/dataTreeWidget.ts [hunk 2]
```diff
@@ -403,6 +404,6 @@ export const generateDataTreeWidget = (
       componentWidth,
       type: widget.type,
     },
-    configEntity: { ...entityConfig, widgetId: dataTreeWidget.widgetId },
+    configEntity: entityConfig,
```
Comment (Rishabh-Rathod) [inline app/client/src/entities/DataTree/dataTreeWidget.ts:407]: Why this change?

widgetId was already present in the entityConfig of the data tree, but we were adding widgetId to the entityConfig later due to a type error. After this change, we could directly pass entityConfig.

File: app/client/src/ce/entities/DataTree/types.ts [hunk 3]
```diff
@@ -163,6 +163,7 @@ export interface WidgetEntityConfig
   extends Partial<WidgetProps>,
     Omit<WidgetConfigProps, "widgetName" | "rows" | "columns">,
     WidgetConfig {
+  widgetId: string;
```
Comment (Rishabh-Rathod) [inline app/client/src/ce/entities/DataTree/types.ts:166]: Why this change?

Check the comment here https://github.com/appsmithorg/appsmith/pull/28752/files#r1387941375

File: app/client/src/ce/utils/getEntityPayloadInfo.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  type WidgetEntity,
+  type WidgetEntityConfig,
+} from "@appsmith/entities/DataTree/types";
+import type { PluginType } from "entities/Action";
+
+export const getEntityPayloadInfo: Record<
+  string,
+  (
+    entity: any,
+    entityConfig: any,
```
Comment (ankitakinger) [inline app/client/src/ce/utils/getEntityPayloadInfo.ts:16]: can we use better types here instead of `any`?

Comment (Rishabh-Rathod) [reply app/client/src/ce/utils/getEntityPayloadInfo.ts:16]: I tried using the `DataTreeEntity` and `DataTreeEntityConfig` but they don't work here due to the way we have implemented the types currently. The union of `CE_DataTreeEntity` and `EE_DataTreeEntity` doesn't accept `WidgetEntity` as type though `CE_DataTreeEntity` is union of `WidgetEntity | ActionEntity | JSActionEntity`.  I tried `unknown` too but also doesn't work. 
image

File: app/client/src/ce/utils/getEntityPayloadInfo.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  ENTITY_TYPE_VALUE,
+} from "@appsmith/entities/DataTree/types";
+import type { PluginType } from "entities/Action";
+
+export const getEntityPayloadInfo: Record<
+  string,
+  (entityConfig: EntityConfig) => {
+    iconId: string;
+    id: string;
+    pluginType?: PluginType;
```
Comment (ankitakinger) [inline app/client/src/ce/utils/getEntityPayloadInfo.ts:17]: I hope this works fine on EE and doesn't break anything.

Comment (Rishabh-Rathod) [reply app/client/src/ce/utils/getEntityPayloadInfo.ts:17]: Yes, this is only used in evalErrorHandler which is introduced in this PR. I don't foresee a reason for this to break anything.

**Review by ankitakinger — APPROVED** [review]
LGTM. @ApekshaBhosale Please review this PR from your end too once.

**Review by ApekshaBhosale — APPROVED** [review]

File: app/client/src/sagas/PostEvaluationSagas.ts [hunk 6]
```diff
@@ -62,276 +48,6 @@ import type {
...
-      const idField = isWidget(entity) ? entity.widgetId : entity.actionId;
-      const nameField = isWidget(entity) ? entity.widgetName : entityName;
-      const entityType = isWidget(entity)
-        ? ENTITY_TYPE.WIDGET
-        : isAction(entity)
-        ? ENTITY_TYPE.ACTION
-        : ENTITY_TYPE.JSACTION;
-      const pluginTypeField = isAction(entity)
-        ? entityConfig.pluginType
-        : entity.type;
```
Comment (Rishabh-Rathod) [inline app/client/src/sagas/PostEvaluationSagas.ts:125]: @ApekshaBhosale This is old code where we were already sending string earlier but it was not strictly typed code.

**Pull request merged**